adding dai warning for erc20 token vs xdai native token

brossetti1 opened the pull request:
in reference [to this discussion](https://discordapp.com/channels/684227450204323876/744945190401474710/757340078354727044)

I opened an [issue here](https://github.com/raid-guild/xdai-omnibridge/issues/82)

I chose not to make this a seperate component because it is just an alert, and its not very complicated. let me know if you need anything to change here, thanks.

### warning in confirm modal:
<img width="483" alt="Screen Shot 2020-09-20 at 3 26 21 PM" src="https://user-images.githubusercontent.com/5998100/93726464-14f69c00-fb6b-11ea-92a1-1ab3b0433d98.png">

### warning when dai is selected:
<img width="1224" alt="Screen Shot 2020-09-20 at 3 46 01 PM" src="https://user-images.githubusercontent.com/5998100/93726465-158f3280-fb6b-11ea-990d-064dd2dd7f9b.png">

dan13ram commented on packages/react-app/src/components/ConfirmTransferModal.jsx:
```diff
@@ -34,6 +36,7 @@ export const ConfirmTransferModal = ({ isOpen, onClose }) => {
   const fromUnit = fromToken.symbol + (isxDai ? ' on xDai' : '');
   const toAmt = formatValue(toAmount, toToken.decimals);
   const toUnit = toToken.symbol + (isxDai ? '' : ' on xDai');
+  const isERC20Dai = toToken.symbol === 'DAI';
```
In either cases, should we do a hard-coded address check instead of symbol check?

dan13ram reviewed (changes requested):
Looks great @brossetti1 
Please add this to the responsive mobile screen as well and we can merge this :D

dan13ram commented on packages/react-app/src/components/BridgeTokens.jsx:
```diff
@@ -9,9 +17,25 @@ import { ToToken } from './ToToken';
+const DaiWarning = () => {
+  return (
+    <Flex align="flex-middle" direction="column">
+      <Alert status="warning" borderRadius={5} mb={5}>
```
we can actually move this alert to a separate component and reuse it

dan13ram commented on packages/react-app/src/components/BridgeTokens.jsx:
```diff
@@ -9,9 +17,25 @@ import { ToToken } from './ToToken';
+      <Alert status="warning" borderRadius={5} mb={5}>
+        <AlertIcon />
+        <Text fontSize="small">
+          Bridges erc20 Dai to xDai, not the xDai Native token.
```
Lets wait for xDai team to approve the text as well.

brossetti1 commented:
@dan13ram I took another look at this one, split the `DaiWarning` into its own component, and added address checking. One of the scenarios that popped up from that is I had to add logic for which network the user is connected to. since the [content of the message](https://discordapp.com/channels/684227450204323876/744945190401474710/757836994586869760) from the xdai team is:

```
Bridges DAI on Ethereum to DAI on xDai, DOES NOT mint native xDai. If you want native xDai, use the <a href=“https://dai-bridge.poa.network/”>ExDai Bridge</a>.
``` 

I opted to restrict the `<DaiWarning /> to dai fromToken's on ethereum based networks` as the message does not apply going from xdai to dai.

I added the dai address constant in the DaiWarning component as I didn't want to clutter the constants only to store Dai addresses. Let me know what you think or if you want changes here.

### Selection process on large screen:
![Screen Shot 2020-09-22 at 8 03 47 PM](https://user-images.githubusercontent.com/5998100/93964101-0a2f3900-fd1c-11ea-9f90-1fb8ff7fd0b1.png)
![Screen Shot 2020-09-22 at 8 05 12 PM](https://user-images.githubusercontent.com/5998100/93964103-0ac7cf80-fd1c-11ea-8be9-372ad161438f.png)

### Selection process on small screen:
![Screen Shot 2020-09-22 at 8 11 51 PM](https://user-images.githubusercontent.com/5998100/93964106-0b606600-fd1c-11ea-9ebb-4073e277908b.png)
![Screen Shot 2020-09-22 at 8 12 12 PM](https://user-images.githubusercontent.com/5998100/93964107-0bf8fc80-fd1c-11ea-976b-77bf841148ea.png)

brossetti1 commented on packages/react-app/src/components/DaiWarning.jsx:
```diff
@@ -0,0 +1,38 @@
+import { Flex, Alert, AlertIcon, Text, Link } from "@chakra-ui/core";
+import { ETEHEREUM_NETWORKS } from "../lib/constants";
+
+const ERC20DaiAddress = {
+  100: '0x44fA8E6f47987339850636F88629646662444217',
+  1: '0x6B175474E89094C44Da98b954EedeAC495271d0F',
+  42: '0x4F96Fe3b7A6Cf9725f59d353F723c1bDb64CA6Aa',
+  77: '0xa844e8c64608dede1f22f519ad0e98e2629684df'
+}
```
a little unconventional but I dont see this information being used outside this component.

brossetti1 replied:
a little unconventional but I dont see this information being used outside this component.

brossetti1 commented on packages/react-app/src/components/DaiWarning.jsx:
```diff
@@ -0,0 +1,38 @@
+  77: '0xa844e8c64608dede1f22f519ad0e98e2629684df'
+}
+
+export const isERC20DaiAddress = (token) => {
+  if (!token) { return false; }
+
+  const isEthNetwork = ETEHEREUM_NETWORKS.includes(token.chainId);
+  return isEthNetwork && token.address === ERC20DaiAddress[token.chainId];
+}
```
this is also a little unconventional and it is exported, but I only see this being used WITH this component so I put it here.

brossetti1 replied:
this is also a little unconventional and it is exported, but I only see this being used WITH this component so I put it here.

brossetti1 commented on packages/react-app/src/lib/constants.jsx:
```diff
@@ -4,6 +4,7 @@ import { CONFIG } from '../config';
 import { NetworkIcon } from '../icons/NetworkIcon';
 
 export const ADDRESS_ZERO = '0x0000000000000000000000000000000000000000';
+export const ETEHEREUM_NETWORKS = [1, 42];
```
¯\_(ツ)_/¯

scottrepreneur replied:
got an extra e in ETHEREUM_NETWORKS

dan13ram replied:
Hey instead of creating this new variable there's an existing method called isxDaiChain in `lib/helpers` which helps you do the same thing.

dan13ram commented on packages/react-app/src/components/DaiWarning.jsx:
```diff
@@ -0,0 +1,38 @@
+            color="blue.500"
+            rel="noreferrer noopener"
+            isExternal>
+            ExDai Bridge
```
It should be `xDai Bridge` not `ExDai`.
Also you don't need to set `rel="..."` when you are setting isExternal.

dan13ram commented on packages/react-app/src/components/BridgeTokens.jsx:
```diff
@@ -1,7 +1,16 @@
-import { Flex, Grid, Text, useBreakpointValue } from '@chakra-ui/core';
+import {
+  Alert,
+  AlertIcon,
```
This is causing lint errors, please remove it. Else it won't build.

dan13ram commented on packages/react-app/src/components/DaiWarning.jsx:
```diff
@@ -0,0 +1,38 @@
+  return (
+    <Flex align="flex-middle" direction="column">
+      <Alert status="warning" borderRadius={5} mb={5}>
+        <AlertIcon />
```
Can you set a minW for this icon, it becomes too small for smaller screens

dan13ram reviewed (changes requested):
Great stuff man! Just some small changes before we can merge it :)

brossetti1 commented:
@dan13ram nice catch, fixed the remaining feedback. heres a shot of the small screen alert icon

<img width="464" alt="Screen Shot 2020-09-23 at 8 03 20 AM" src="https://user-images.githubusercontent.com/5998100/94023412-68d3d180-fd73-11ea-9d56-884f8d4c7239.png">

scottrepreneur reviewed (approved):
great work
